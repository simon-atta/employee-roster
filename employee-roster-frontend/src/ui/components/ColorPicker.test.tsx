/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { shallow } from 'enzyme';
import { Popover } from '@patternfly/react-core';
import { ColorPickerProps, ColorPicker } from './ColorPicker';

describe('ColorPicker component', () => {
  const baseProps: ColorPickerProps = {
    currentColor: '#ff0000',
    onChangeColor: jest.fn(),
  };
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const colorPicker = shallow(<ColorPicker {...baseProps} />);
    expect(colorPicker).toMatchSnapshot();
  });

  it('should set color on click', () => {
    const colorPicker = shallow(<ColorPicker {...baseProps} />);
    const newColor = '#0000ff';
    shallow(colorPicker.find(Popover).prop('bodyContent') as React.ReactElement)
      .simulate('change', newColor);

    expect(baseProps.onChangeColor).toBeCalledWith(newColor);
  });
});
