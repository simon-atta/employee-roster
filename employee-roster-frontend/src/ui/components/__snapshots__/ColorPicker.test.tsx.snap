// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component should render correctly 1`] = `
<Popover
  aria-label="Select Color..."
  bodyContent={
    <ColorPicker
      color="#ff0000"
      colors={
        Array [
          "#7D1007",
          "#A30000",
          "#C9190B",
          "#2B9AF3",
          "#73BCF7",
          "#BEE1F4",
          "#F4C145",
          "#F6D173",
          "#F9E0A2",
          "#EC7A08",
          "#EF9234",
          "#F4B678",
          "#6EC664",
          "#95D58E",
          "#BDE5B8",
          "#A18FFF",
          "#B2A3FF",
          "#CBC1FF",
          "#A2D9D9",
          "#73C5C5",
          "#009596",
        ]
      }
      onChange={[Function]}
      styles={Object {}}
      triangle="hide"
      width={170}
    />
  }
  isVisible={false}
  shouldClose={[Function]}
>
  <Button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ff0000",
        "color": "white",
        "width": "100%",
      }
    }
  >
    #ff0000
  </Button>
</Popover>
`;
